### ISSUE TITLE
Handling @JsonValue/@JsonCreator

### ISSUE THREAD

**Issue by sortel64** [vojtechhabarta/typescript-generator#560]
Given a Java class that is annotated with the Jackson2 @JsonCreator/@JsonValue, the resulting interface/class is empty. I'm not good enough with typescript to know if this is correct, but it seems not to be.

For example given the Java class (mine is more complicated):
```
public class URN {
    private String nid;
    private String nss;

    @JsonCreator
    public URN(String str) {
       this.nid = parseNid(str);
       this.nss = parseNss(str);
    }

    @JsonValue
    public String toString() {
        return "urn:" + nid + ":" + nss;
    }
}
```
The generated typescript class is:
```
export class URN {
}
```
Given that the return type of the function annotated with @JsonValue and single parameter on the @JsonCreator constructor, the class can be determined to be assignment compatible with a String.  Therefore the generated code could simply be:

`type URN = string;`

Again, I'm not a wizard with typescript and would appreciate any sight you can provide.
Thank you for your consideration.

**Comment by vojtechhabarta** [comment]
You are right. Type of the property should be either `URN` where `type URN = string;` or it can be directly `string`.

Typescript-generator should handle this case but it currently does not.
But it is possible to use [customTypeMappings](https://www.habarta.cz/typescript-generator/maven/typescript-generator-maven-plugin/generate-mojo.html#customTypeMappings) parameter to map `org.package.URN` to `string` manually. Or you can map it to `URN` TypeScript type and define this type using [customTypeAliases](https://www.habarta.cz/typescript-generator/maven/typescript-generator-maven-plugin/generate-mojo.html#customTypeAliases) parameter.

**Comment by sortel64** [comment]
Thanks for the consideration and response.  I went the customTypeMapping route and that worked well for the few cases that I have now.

**Comment by renedewaele** [comment]
hi @vojtechhabarta, is there an update planned for this? At the moment we have quite a few custom mappings for this in our project. It would be **really** nice if the plugin would support this automatically via `@JsonValue`. 🙂
